bin/generate_east_asian_width

```diff
@@ -5,6 +5,18 @@ if ARGV.empty?
   exit 1
 end
 
+def unicode_width(type, category)
+  return 0 if category == 'Mn' # Nonspacing Mark
```

Comment:
Is there any reason why the Nonspacing Mark should be set to 0?

Reply:
In the old implementation, Mark(`Mn (nonspacing mark) + Mc (spacing combining mark) + Me (enclosing mark)`) are all set to 0.

```ruby
MBCharWidthRE = /
  ...
  | (?<width_0>^\p{M})
  ...
/
```

I think this is just a simple mistake of `\p{Mn}` because Mark is not always zero width.
I think there are two choices and I choose the latter one.
```ruby
# Choice 1, Don't change the behavior now. Leave it, or fix it in another pull request
return 0 if category =~ '/Mn|Mc|Me/'
# Choice 2, Fix it to nonspacing
return 0 if category == 'Mn'
```

Actual width calculated by the script below are:
| Terminal emulator | Mn | Other marks(Mc, Me) |
| --- | -- | -- |
| Terminal.app | 0 | >=1 |
| Alacritty | 0 | mostly(97%) >=1, some(3%) are 0 |
| iTerm, VSCode Terminal | 1 | >= 1 |

```ruby
mn = (0..0x10ffff).filter_map{_1.chr('utf-8') rescue nil}.grep(/\p{Mn}/)
mn.map{
  print "\e[H"+_1+"\e[6n"
  STDIN.raw{STDIN.readpartial 10}
}.tally
```
Mn is not zero-width in some terminal emulators, but I prefer not changing the original intention in this pull request except bug.

Reply:
Thanks for your explanation. That makes sense to me now.